module/zfs/dmu_send.c

```diff
@@ -2315,6 +2333,25 @@ dmu_send_impl(struct dmu_send_params *dspp)
 		dsl_pool_rele(dp, tag);
 		return (err);
 	}
+
+	/*
+	 * If we're sending partial state resume fields must exist.
+	 */
+	if (dspp->partialok) {
+		struct drr_begin drrb;
+		objset_t *mos = dmu_objset_pool(os)->dp_meta_objset;
+		err = zap_lookup(mos, dmu_objset_id(os),
+		    DS_FIELD_RESUME_TOGUID, sizeof (drrb.drr_toguid), 1,
+		    &drrb.drr_toguid);
```

Comment:
Let's keep it simple and just return on error immediately rather than or'ing together the errors.  This way we'll be able to trace which case we hit in the event of an error, and it's consistent with the existing style.